src/dev/build/tasks/fetch_agent_versions_list.ts

```diff
@@ -44,6 +60,11 @@ export const FetchAgentVersionsList: Task = {
   description: 'Build list of available Elastic Agent versions for Fleet UI',
 
   async run(config, log, build) {
+    // Agent version list task is skipped for PR's, so as not to overwhelm the versions API
+    if (process.env.BUILDKITE_PULL_REQUEST === 'true') {
+      return;
+    }
+
```

Comment:
To avoid hammering this product versions API, which has some stability issues (actively being worked on by @brianjolly and his team 🙏), we'll skip building this versions list in PR's. This has a very minor inconvenience in cloud-first testing for PR's but generally won't impact developer in a meaningful way, so it makes sense to save on traffic here and avoid blowing up that API.